### PR TITLE
features/ui_rebuild story 105 and 109:  http client and login page

### PR DESCRIPTION
**Pull request by leefrank9527**
Implemented the story https://github.com/WebCuratorTool/webcurator/issues/105 and story https://github.com/WebCuratorTool/webcurator/issues/109:

Imported the components and services of primevue and primeicon.
Implemented a http client for convienient usage.
Implemented the login page and authentication, and integrated with the http client.
Added an example (Dashboard.vue) for how to call the http client.
Added the context path for routes, so the navigation of compiled webjars can do work.

### CODE REVIEW

File: webcurator-ui/src/components/LoginDialog.vue [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// import { useToast } from 'primevue/usetoast';
+// const toast = useToast();
+
+const dialogRef:any = inject("dialogRef");
+const username=ref(null);
+const password=ref(null);
+
+
+const auth = () => {
+    var url="./auth/v1/token?username=" + username.value + "&password=" + password.value;
```
Comment (obrienben) [inline webcurator-ui/src/components/LoginDialog.vue:41]: @leefrank9527 is this putting a password in the query string that could be seen in the HTTP headers?

Comment (leefrank9527) [reply webcurator-ui/src/components/LoginDialog.vue:41]: > @leefrank9527 is this putting a password in the query string that could be seen in the HTTP headers?

Yes. The credential is included in the URL, I would put it inside the post body instead.

Comment (obrienben) [reply webcurator-ui/src/components/LoginDialog.vue:41]: Ok. And its already encrypted right?

Comment (leefrank9527) [reply webcurator-ui/src/components/LoginDialog.vue]: > Ok. And its already encrypted right?

It dependends on the transfer protocol. If the https is used, the credential in the http body will be encryped. If the http is used, the credential is still plain text. 

Have updated the branch to put the credential in the http body.

**Pull request merged**